test/ldap_test.rb

```diff
@@ -117,12 +118,79 @@ def test_search_strategy_misconfigured_to_unrecognized_strategy_falls_back_to_de
   def test_capabilities
     assert_kind_of Net::LDAP::Entry, @ldap.capabilities
   end
+
+  def test_global_catalog_connection_is_null_if_not_active_directory
+    @ldap.expects(:active_directory_capability?).returns(false)
+    assert_equal nil, @ldap.global_catalog_connection
+  end
+
+  def test_global_catalog_connection_is_null_if_not_active_directory
+    @ldap.expects(:active_directory_capability?).returns(true)
+    refute_nil @ldap.global_catalog_connection, "Expected Global Catalog to not be nil"
```

Comment:
nitpick, but I'd drop the message since the `refute_nil` is already pretty self explanatory